projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -93,6 +93,47 @@ export const mergeObjects = (obj1: any, obj2: any): any => mergeWith(obj1, obj2,
     }
 });
 
+/**
+ * Recursively checks if two objects are deeply equal.
+ * Handles circular references by keeping track of seen objects.
+ *
+ * @param obj1 First object to compare
+ * @param obj2 Second object to compare
+ * @param seen Set of already visited objects to handle circular references
+ * @returns true if objects are equal, false otherwise
+ * @hidden
+ */
+export const areObjectsEqual = (obj1: any, obj2: any, seen = new Set()): boolean => {
```

Comment:
We don't need this function - we have a dependency to `lodash-es` which exposes an [isEqual](https://lodash.com/docs#isEqual) function which is pretty good at deep object comparisons.